src/wallet/rpcnames.cpp

```diff
@@ -483,11 +523,81 @@ name_firstupdate ()
   if (name.size () > MAX_NAME_LENGTH)
     throw JSONRPCError (RPC_INVALID_PARAMETER, "the name is too long");
 
-  const valtype rand = ParseHexV (request.params[1], "rand");
-  if (rand.size () > 20)
-    throw JSONRPCError (RPC_INVALID_PARAMETER, "invalid rand value");
+  const bool fixedRand = request.params.size () >= 2 && !request.params[1].isNull ();
+  const bool fixedTxid = request.params.size () >= 3 && !request.params[2].isNull ();
```

Comment:
Nit:  I personally would wrap the right-hand side of the assignment into parentheses for expressions like this, but that is optional.